File: output/GPT3.5-Turbo/xCodeEval_remove_4000/C++/Python/18870ef0a72bdee24f7c568674dc1c2d.py
import math

SQ = 100 * 1000
N = 3e8 + SQ
l, r = map(int, input().split())

def prime(n):
    for i in range(2, int(math.sqrt(n)) + 1):
        if n % i == 0:
            return False
    return n > 1

def check(n):
    if n == 2:
        return True
    if n % 4 != 1:
        return False
    return prime(n)

def calc(l, r):
    pl = l // SQ
    pr = r // SQ
    ans = 0

    if pl == pr:
        for i in range(l, r + 1):
            if check(i):
                ans += 1
        return ans

    while l < (pl + 1) * SQ:
        if check(l):
            ans += 1
        l += 1
    while r >= pr * SQ:
        if check(r):
            ans += 1
        r -= 1

    for i in range(pl + 1, pr):
        ans += res[i]

    return ans

print(calc(l, r))

File: output/GPT3.5-Turbo/xCodeEval_4000/C++/Go/18870ef0a72bdee24f7c568674dc1c2d.go
package main

import (
	"fmt"
	"math"
)

const SQ = 100 * 1000
const N = 3e8 + SQ

var l, r int
var res [N/SQ + 1]int

func prime(n int) bool {
	for i := 2; i*i <= n; i++ {
		if n%i == 0 {
			return false
		}
	}
	return n > 1
}

func check(n int) bool {
	if n == 2 {
		return true
	}
	if n%4 != 1 {
		return false
	}
	return prime(n)
}

func calc(l, r int) int {
	pl := l / SQ
	pr := r / SQ
	ans := 0

	if pl == pr {
		for i := l; i <= r; i++ {
			if check(i) {
				ans++
			}
		}
		return ans
	}

	for l < (pl+1)*SQ {
		if check(l) {
			ans++
		}
		l++
	}
	for r >= pr*SQ {
		if check(r) {
			ans++
		}
		r--
	}

	for i := pl + 1; i < pr; i++ {
		ans += res[i]
	}

	return ans
}

func main() {
	fmt.Scan(&l, &r)

	fmt.Println(calc(l, r))
}
